src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentsState = { ...state };
+  const stateHistory = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        currentsState = {};
+        break;
+      case 'addProperties':
+        currentsState = { ...currentsState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        currentsState = { ...currentsState };
+
+        action.keysToRemove.forEach((key) => {
+          delete currentsState[key];
+        });
+        break;
+    }
```

Comment:
A `default` case should be added in the `switch` statement to handle any unexpected `action.type`. This is important for error handling and to ensure that the function behaves predictably even with invalid input. [CODE STYLE]